### DISCUSSION TITLE
rez-build and CMAKE_BUILD_TYPE

### DISCUSSION THREAD

**Discussion by sanikache** [AcademySoftwareFoundation/rez#1437] [General]
Hello,

Is there a way for rez-build to read from CMAKE_BUILD_TYPE variable set in a cmake module? rez-build doesn't seem to recognize it when I set it to Debug. It always sets it to -DCMAKE_BUILD_TYPE=Release. I am trying to avoid passing the `--bt` flag.

Thanks.

**Comment by predat** [comment]
Hi @sanikache,

We, at the sudio, use a rez package "build" with 2 versions "release" and "debug".
Each one contains an environment variable env.CMAKE_BUILD_TYPE = "Release" or "Debug".


Then the packages that have to be compiled have "build-release" and "build-debug" variants. Example:
```python
variants = [
  ["platform-linux", "build-release"],
  ["platform-linux", "build-debug"]
]
```

**Reply by sanikache** [reply]
Thanks @predat! We have a similar variants structure with build_type-debug and build_type-release packages that does exactly what you described and works with our SCons build system. I think the change that I am looking for is in this line here: https://github.com/AcademySoftwareFoundation/rez/blob/c0cc4bf1542a6d7a7b41fe5871e28743c1cd0f6b/src/rezplugins/build_system/cmake.py#L87
Setting the default to a value being set by ${CMAKE_BUILD_TYPE} if it exists, otherwise fallback to settings.build_target. That should allow rez-build to pick up the CMAKE_BUILD_TYPE being set by the build environment. I will test it in a custom rezplugin to see if that works.
